jgiven-core/src/main/java/com/tngtech/jgiven/report/asciidoc/AsciiDocReportGenerator.java

```diff
@@ -1,196 +1,186 @@
-
-        allFiles.add( targetFileName );
-        if( !config.getTargetDir().exists() ) {
-            config.getTargetDir().mkdirs();
-        }
-        File targetFile = new File( config.getTargetDir(), targetFileName );
-        PrintWriter printWriter = PrintWriterUtil.getPrintWriter( targetFile );
-
-        try {
-            new AbstractReportModelHandler().handle( model, new AsciiDocReportModelVisitor( printWriter ) );
-        } finally {
-            ResourceUtil.close( printWriter );
-        }
+    public AbstractReportConfig createReportConfig(String... args) {
+        return new AsciiDocReportConfig(args);
     }
 
-    private void generateIndexFile() {
-        PrintWriter printWriter = PrintWriterUtil.getPrintWriter( new File( config.getTargetDir(), "index.asciidoc" ) );
-        try {
-            printWriter.println( "= JGiven Documentation =\n" );
-            printWriter.println( ":toc: left\n" );
-            printWriter.println( "== Scenarios ==\n" );
-            printWriter.println( "=== Classes ===\n" );
-            printWriter.println( "include::allClasses.asciidoc[]" );
-        } finally {
-            ResourceUtil.close( printWriter );
-        }
-
-        printWriter = PrintWriterUtil.getPrintWriter( new File( config.getTargetDir(), "allClasses.asciidoc" ) );
-        try {
-            for( String fileName : allFiles ) {
-                printWriter.println( "include::" + fileName + "[]\n" );
-            }
-        } finally {
-            ResourceUtil.close( printWriter );
-
+    @Override
+    public void generate() {
+        File targetDir = config.getTargetDir();
+        if (!targetDir.exists() && !targetDir.mkdirs()) {
+            log.error("Could not ensure target directory exists {}", targetDir);
+            return;
         }
 
-    }
-
-    class AsciiDocReportModelVisitor implements ReportModelHandler {
-
-        private final PrintWriter writer;
-
-        AsciiDocReportModelVisitor( PrintWriter printWriter ) {
-            this.writer = printWriter;
+        featuresDir = new File(targetDir.getPath() + "/features");
+        if (!featuresDir.exists() && !featuresDir.mkdirs()) {
+            log.error("Could not ensure feature directory exists {}", featuresDir);
+            return;
         }
 
-        @Override
-        public void className( String className ) {
-            writer.println( "==== " + className + " ====\n" );
-        }
+        completeReportModel.getAllReportModels().stream()
+            .sorted(byFeatureName)
+            .forEach(reportModelFile -> writeFeatureToFile(reportModelFile.model, reportModelFile.file,
+                completeReportModel.getStatistics(reportModelFile)));
 
-        @Override
-        public void reportDescription( String description ) {
-            writer.println( description );
-            writer.println();
-        }
+        try (PrintWriter printWriter = PrintWriterUtil.getPrintWriter(new File(targetDir, "allScenarios.asciidoc"))) {
```

Comment:
could these `try` statements be split into several methods, like `writeIndex`, `writeGlobalStatistics` etc.?